core/src/test/java/org/apache/iceberg/rest/requests/TestCreateNamespaceRequest.java

```diff
@@ -0,0 +1,129 @@
+  private static final Map<String, String> EMPTY_PROPERTIES = ImmutableMap.of();
+
+  @Test
+  // Test cases that are JSON that can be created via the Builder
+  public void testRoundTripSerDe() throws JsonProcessingException {
+    String fullJson = "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":{\"owner\":\"Hank\"}}";
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE).setProperties(PROPERTIES).build();
+    assertRoundTripSerializesEquallyFrom(fullJson, req);
+
+    String jsonEmptyProperties = "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":{}}";
+    CreateNamespaceRequest reqWithExplicitEmptyProperties = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE).setProperties(EMPTY_PROPERTIES).build();
+    assertRoundTripSerializesEquallyFrom(jsonEmptyProperties, reqWithExplicitEmptyProperties);
+
+    CreateNamespaceRequest reqWithImplicitEmptyProperties = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE).build();
+    assertRoundTripSerializesEquallyFrom(jsonEmptyProperties, reqWithImplicitEmptyProperties);
+  }
+
+  @Test
+  // Test cases that can't be constructed with our Builder class but that will parse correctly
+  public void testCanDeserializeWithoutDefaultValues() throws JsonProcessingException {
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder().withNamespace(NAMESPACE).build();
+    String jsonWithNullProperties = "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":null}";
+    assertEquals(deserialize(jsonWithNullProperties), req);
+
+    String jsonWithMissingProperties = "{\"namespace\":[\"accounting\",\"tax\"]}";
+    assertEquals(deserialize(jsonWithMissingProperties), req);
+  }
+
+  @Test
+  public void testDeserializeInvalidRequest() {
+    String jsonIncorrectTypeForNamespace = "{\"namespace\":\"accounting%00tax\",\"properties\":null}";
+    AssertHelpers.assertThrows(
+        "A JSON request with incorrect types for fields should fail to deserialize and validate",
+        JsonProcessingException.class,
+        "Cannot parse string array from non-array value",
+        () -> deserialize(jsonIncorrectTypeForNamespace).validate()
+    );
+
+    String jsonIncorrectTypeForProperties = "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":[]}";
+    AssertHelpers.assertThrows(
+        "A JSON request with incorrect types for fields should fail to parse and validate",
+        JsonProcessingException.class,
+        () -> deserialize(jsonIncorrectTypeForProperties).validate()
+    );
+
+    String jsonMisspelledKeys = "{\"namepsace\":[\"accounting\",\"tax\"],\"propertiezzzz\":{\"owner\":\"Hank\"}}";
+    AssertHelpers.assertThrows(
+        "A JSON request with the keys spelled incorrectly should fail to deserialize and validate",
+        JsonProcessingException.class,
+        () -> deserialize(jsonMisspelledKeys).validate()
```

Comment:
Would be nice to have error message validation.

Reply:
I've added in error message validation. But I didn't put it in some places where the error message comes form Jackson. I can do so if you'd like.